src/ui/public/timepicker/timepicker.js

```diff
@@ -124,21 +153,33 @@ module.directive('kbnTimepicker', function (quickRanges, timeUnits, refreshInter
         $scope.absolute.to = moment();
       };
 
-      $scope.formatRelative = function () {
-        let parsed = dateMath.parse(getRelativeString());
-        $scope.relative.preview =  parsed ? parsed.format($scope.format) : undefined;
+      $scope.formatRelativeFrom = function () {
+        var parsed = dateMath.parse(getRelativeFromString());
+        $scope.relative.from.preview = parsed ? parsed.format($scope.format) : undefined;
+        return parsed;
+      };
+
+      $scope.formatRelativeTo = function () {
+        var parsed = dateMath.parse(getRelativeToString());
```

Comment:
This and the preceding function is the same, can you try, and make a function that perhaps returns a function that can do these operations?